pkg/dbsyncers/policy_syncer.go

```diff
@@ -149,10 +151,87 @@ func updateComplianceStatus(ctx context.Context, k8sClient client.StatusClient,
 		policy.Status.ComplianceState = policiesv1.Compliant
 	}
 
-	err := k8sClient.Status().Patch(ctx, policy, client.MergeFrom(originalPolicy))
+	placements, err := getPlacements(k8sClient, policy)
+	if err != nil {
+		return err
+	}
+	policy.Status.Placement = placements
+
+	err = k8sClient.Status().Patch(ctx, policy, client.MergeFrom(originalPolicy))
 	if err != nil && !errors.IsNotFound(err) {
 		return fmt.Errorf("failed to update policy CR: %w", err)
 	}
 
 	return nil
 }
+
+func getPlacements(k8sClient client.Client, instance *policiesv1.Policy) ([]*policiesv1.Placement, error) {
+	log.Info("get the placements for the policy", "policy", instance.GetName())
+	pbList, err := getPlacementBindingList(k8sClient, instance)
+	if err != nil {
+		return nil, err
+	}
+	var placements []*policiesv1.Placement
+	for _, pb := range pbList.Items {
+		subjects := pb.Subjects
+		for _, subject := range subjects {
+			if !(subject.APIGroup == policiesv1.SchemeGroupVersion.Group &&
+				subject.Kind == policiesv1.Kind &&
+				subject.Name == instance.GetName()) {
+				continue
+			}
+			p, err := getPlacementsPerBinding(k8sClient, pb, instance)
+			if err != nil {
+				return nil, err
+			}
+			placements = append(placements, p...)
+		}
+
+	}
+	return placements, nil
+}
+
+func getPlacementBindingList(k8sClient client.Client, instance *policiesv1.Policy) (*policiesv1.PlacementBindingList, error) {
+	// Get the placement binding in order to later get the placements
+	pbList := &policiesv1.PlacementBindingList{}
+
+	err := k8sClient.List(context.TODO(), pbList, &client.ListOptions{Namespace: instance.GetNamespace()})
+	if err != nil {
+		log.Error(err, "failed to list placementbindings")
+		return nil, err
+	}
+	return pbList, nil
+}
+
+// getPlacementsPerBinding returns the placements for the policy per placementbinding
+func getPlacementsPerBinding(k8sClient client.Client, pb policiesv1.PlacementBinding,
+	instance *policiesv1.Policy) ([]*policiesv1.Placement, error) {
+	plr := &placementrulesv1.PlacementRule{}
+
+	err := k8sClient.Get(context.TODO(), types.NamespacedName{
+		Namespace: instance.GetNamespace(),
+		Name:      pb.PlacementRef.Name,
+	}, plr)
+	// no error when not found
+	if err != nil && !errors.IsNotFound(err) {
+		return nil, fmt.Errorf("failed to get placementrule: %w", err)
+	}
+
+	var placements []*policiesv1.Placement
+
+	plcPlacementAdded := false
+
+	for _, subject := range pb.Subjects {
+		if subject.Kind == policiesv1.Kind && subject.Name == instance.GetName() && !plcPlacementAdded {
+			placement := &policiesv1.Placement{
+				PlacementBinding: pb.GetName(),
+				PlacementRule:    plr.GetName(),
+			}
+			placements = append(placements, placement)
+			// should only add policy placement once in case placement binding subjects contains duplicated policies
+			plcPlacementAdded = true
```

Comment:
why not using `break` instead of this boolean?
e.g. something like:
```
if subject.Kind != policiesv1.Kind || subject.Name != instance.GetName() {
    continue
}
// kind and name match the policy
placement := &policiesv1.Placement{
	PlacementBinding: pb.GetName(),
	PlacementRule:    plr.GetName(),
}
placements = append(placements, placement)
break // should only add policy placement once in case placement binding subjects contains duplicated policies
```

Reply:
Yes. that makes sense. the `PlacementBinding` can refer `placementrule` to multiple policies. but here we only handle one policy. so `break` is good. Thanks.